mithril-aggregator/src/multi_signer.rs

```diff
@@ -284,15 +293,38 @@ impl MultiSignerImpl {
         protocol_parameters: &ProtocolParameters,
     ) -> Result<Option<ProtocolClerk>, ProtocolError> {
         debug!("Create clerk");
-        let mut key_registration = ProtocolKeyRegistration::init();
+        let stake_distribution = signers_with_stake
+            .iter()
+            .map(|s| s.into())
+            .collect::<ProtocolStakeDistribution>();
+        let mut key_registration = ProtocolKeyRegistration::init(&stake_distribution);
         let mut total_signers = 0;
         for signer in signers_with_stake {
-            if let Ok(Some(verification_key)) = key_decode_hex(&signer.verification_key) {
-                key_registration
-                    .register(signer.stake, verification_key)
-                    .map_err(|e| ProtocolError::Core(e.to_string()))?;
-                total_signers += 1;
-            }
+            let operational_certificate = match &signer.operational_certificate {
+                Some(operational_certificate) => {
+                    key_decode_hex(operational_certificate).map_err(ProtocolError::Codec)?
```

Comment:
This could be done, but when we will enforce the keys certification we will not be able to register a signer without a valid `Operational Certificate` in `ProtocolKeyRegistration::register`? WDYT?